docs(nx-cloud): add info about start-ci-run and --stop-agents-after u… (#22888)
nrwl/nx: docs/nx-cloud/reference/nx-cloud-cli.md

At the beginning of your main job, invoke `npx nx-cloud start-ci-run`. This tells Nx Cloud that the following series of
command correspond to the same CI run.

{% callout type="warning" title="Do not run start-ci-run locally" %}

> `nx-cloud start-ci-run` is designed to run in CI. If you run the command locally, it can cause your local Nx repo think it's apart of a CI run. This can cause strange behavior like Nx commands timing out or throwing unexpected errors.

Reviewer: ```suggestion
`nx-cloud start-ci-run` is designed to run in CI. If you run the command locally, it can cause your local Nx repo to think it's a part of a CI run. This can cause strange behavior like Nx commands timing out or throwing unexpected errors.
```

> If you ran the command locally, you can remove your systems temp files to reset your local system. Typically restarting your system is enough to fix this issue.

Author: @StalkAltan can you confirm this is the correct way to clear the system when start-ci-run is ran locally? couldn't remember if you also needed to nuke node_modules or not.

Reviewer 2: This is fine, and it will be harder to run this locally very soon (requiring a --force flag)

> If you ran the command locally, you can remove your systems temp files to reset your local system. Typically restarting your system is enough to fix this issue.

Reviewer: ```suggestion
If you accidentally run the command locally, you can remove your system's temp files to reset your local system. Typically restarting your system is enough to fix this issue.
```

You can fix it by telling Nx Cloud that it can terminate agents after it sees a certain
target: `npx nx-cloud start-ci-run --stop-agents-after=e2e`.

The target name for `--stop-agents-after` should be the last target ran within your pipeline. If not Nx Cloud will complete the CI pipeline execution, preventing the subsequent commands from running.

For example:

```yaml
- run: npx nx-cloud start-ci-run --stop-agents-after=build

> - run: nx affected -t build,lint,test

Reviewer 2: ```suggestion
- run: nx affected -t build
- run: nx affected -t lint
- run: nx affected -t test
```

```

If build tasks are all cached, then all build tasks will complete immediately causing lint and test tasks to fail with an error saying the CI pipeline execution has already been completed. Instead you should re-order your targets to make sure the build target is last

```yaml
- run: npx nx-cloud start-ci-run --stop-agents-after=build

> - run: nx affected -t lint,test,build

Reviewer 2: ```suggestion
- run: nx affected -t lint
- run: nx affected -t test
- run: nx affected -t build
```

You can fix it by telling Nx Cloud that it can terminate agents after it sees a certain
target: `npx nx-cloud start-ci-run --stop-agents-after=e2e`.

> The target name for `--stop-agents-after` should be the last target ran within your pipeline. If not Nx Cloud will complete the CI pipeline execution, preventing the subsequent commands from running.

Reviewer: ```suggestion
The target name for `--stop-agents-after` should be the last target run within your pipeline. If not, Nx Cloud will end the CI pipeline execution, preventing the subsequent commands from running.
```

> For example:

Reviewer: ```suggestion
Incorrect example:
```

```yaml
- run: npx nx-cloud start-ci-run --stop-agents-after=build
- run: nx affected -t build
- run: nx affected -t lint
- run: nx affected -t test
```

> If build tasks are all cached, then all build tasks will complete immediately causing lint and test tasks to fail with an error saying the CI pipeline execution has already been completed. Instead you should re-order your targets to make sure the build target is last

Reviewer: ```suggestion
If build tasks are all cached, then all build tasks will complete immediately causing lint and test tasks to fail with an error saying the CI pipeline execution has already been completed. Instead you should re-order your targets to make sure the build target is last.
```

> ```yaml

Reviewer: ```suggestion
Corrected example:

```yaml
```
